apps/backend/libs/pipe/src/problem-order.pipe.ts

```diff
@@ -3,16 +3,29 @@ import {
   Injectable,
   type PipeTransform
 } from '@nestjs/common'
-import { ProblemOrder } from '@client/problem/enum/problem-order.enum'
+import type { ProblemOrder } from '@client/problem/enum/problem-order.enum'
 
 @Injectable()
 export class ProblemOrderPipe implements PipeTransform {
-  transform(value: unknown) {
+  transform(value: unknown): ProblemOrder {
     if (!value) {
-      return ProblemOrder.idASC
-    } else if (!Object.values(ProblemOrder).includes(value as ProblemOrder)) {
+      return 'id-asc'
+    }
+    const validValues = [
+      'id-asc',
+      'id-desc',
+      'title-asc',
+      'title-desc',
+      'level-asc',
+      'level-desc',
+      'acrate-asc',
+      'acrate-desc',
+      'submit-asc',
+      'submit-desc'
+    ]
+    if (!validValues.includes(value as string)) {
       throw new BadRequestException('Problem-order validation failed')
     }
```

Comment:
이거는 value가 string 아닌 경우 처리가 예상이 어려워서 type handling도 달아주는 게 좋아보입니다
```suggestion
    if (typeof value !== "string" || !validValues.includes(value)) {
      throw new BadRequestException('Problem-order validation failed')
    }
```